natlib/svcd.c

```diff
@@ -185,3 +185,68 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+static int svcd_unsubscribe( lua_State *L )
+{
+    // check number of args	
+    if (lua_gettop(L) != 4) return luaL_error(L, "Expected (targetip, svcid, attrid, ivkid)");
+    
+    uint16_t svcid = lua_tonumber(L, 2);
+    uint16_t attrid = lua_tonumber(L, 3);
+    uint16_t ivkid = lua_tonumber(L, 4);
+    // set array to var msg
+    
```

Comment:
since these values remain at the bottom of the stack you could use lua_pushvalue(index) to get these numbers instead of storing them in variables. It'll avoid creation of these variables.
PS: I made the same mistake
PB4J